engine/SearchResult.js

```diff
@@ -0,0 +1,20 @@
+// When looking up content within the story (e.g. in Container.ContentAtPath),
+// the result is generally found, but if the story is modified, then when loading
+// up an old save state, then some old paths may still exist. In this case we
+// try to recover by finding an approximate result by working up the story hierarchy
+// in the path to find the closest valid container. Instead of crashing horribly,
+// we might see some slight oddness in the content, but hopefully it recovers!
+export class SearchResult{
+	constructor(){
+		this.obj;
+		this.approximate;
+	}
+
+	get correctObject(){
+		return this.approximate ? null : obj;
+	}
+
+	get container(){
+		return obj;
```

Comment:
So in the [original code](https://github.com/inkle/ink/compare/0.7.5...0.8.1#diff-2c8172fa5b22cfd139d6362cdb03de52R16), `obj` gets casted to a `Container`, which makes it `null` or `falsy` if the cast fails (can't remember which one, but I think it's `null`.)

If we return `obj` as-is, I think we risk returning a non-null value to a calling code would expect `null` under some conditions. So maybe `return this.obj instanceof Container ? this.obj : null` would be preferable here. What do you think?

Reply:
You're totally right. I totally missed the difference between the two types of cast (`(Container)` and `as Container`). I've checked and indeed, `null` would be returned if `obj` wasn't a `Container`. I'll fix it.